addons/binding/org.openhab.binding.loxone/src/main/java/org/openhab/binding/loxone/internal/core/LxControlColorPickerV2.java

```diff
@@ -0,0 +1,174 @@
+        socketClient.sendAction(uuid, "hsv(" + hsb.toString() + ")");
+    }
+
+    /**
+     * Sets the color picker to on
+     *
+     * @throws IOException
+     *                         error communicating with the Miniserver
+     */
+    public void on() throws IOException {
+        HSBType currentColor = this.getColor();
+        if (currentColor != null) {
+            this.setColor(new HSBType(currentColor.getHue(), currentColor.getSaturation(), PercentType.HUNDRED));
+        }
+    }
+
+    /**
+     * Sets the color picker to off
+     *
+     * @throws IOException
+     *                         error communicating with the Miniserver
+     */
+    public void off() throws IOException {
+        HSBType currentColor = this.getColor();
+        if (currentColor != null) {
+            this.setColor(new HSBType(currentColor.getHue(), currentColor.getSaturation(), PercentType.ZERO));
+        }
+    }
+
+    /**
+     * set the brightness level
+     *
+     * @param p the brightness percentage
+     */
+    public void setBrightness(PercentType p) {
+        HSBType currentColor = this.getColor();
+        if (currentColor != null) {
+            this.setColor(new HSBType(currentColor.getHue(), currentColor.getSaturation(), p));
+        }
+    }
+
+    /**
+     * set the brightness level from a decimal type
+     *
+     * @param d the brightness in decimal
+     */
+    public void setBrightness(DecimalType d) {
+        this.setBrightness(new PercentType(d.toBigDecimal()));
+    }
+
+    /**
+     * increases/decreases the brightness with a given step
+     *
+     * @param step the amount to increase/decrease
+     */
+    public void increaseDecreaseBrightness(int step) {
+        HSBType currentColor = this.getColor();
+        if (currentColor != null) {
+            this.setBrightness(new PercentType(currentColor.getBrightness().intValue() + step));
+        }
+    }
+
+    /**
+     * map the Loxone color to OpenHab HSBType
+     *
+     * @param color color in format hsb(h,s,v) or temp(brightness,kelvin)
+     * @return HSBType
+     */
+    private HSBType mapLoxoneToOH(String color) {
+        HSBType hsbColor = null;
+
+        try {
+            Pattern hsvPattern = Pattern.compile("^hsv\\([0-9]\\d{0,2},[0-9]\\d{0,2},[0-9]\\d{0,2}\\)");
+            Pattern tempPattern = Pattern.compile("^temp\\([0-9]\\d{0,2},[0-9]\\d{0,4}\\)");
+            Matcher valueMatcher = Pattern.compile("\\((.*?)\\)").matcher(color);
+
+            if (hsvPattern.matcher(color).matches() && valueMatcher.find()) {
+                // we have a hsv(hue,saturation,value) pattern
+                hsbColor = new HSBType(valueMatcher.group(1));
+            } else if (tempPattern.matcher(color).matches() && valueMatcher.find()) {
+                // we have a temp(brightness,kelvin) pattern
+                hsbColor = TemperatureHSBType.fromBrightnessTemperature(valueMatcher.group(1));
+            }
+        } catch (Exception e) {
```

Comment:
We expect only illegal argument exception, please specify it, so other exceptions don't get hidden.